tawazi/node.py

```diff
@@ -134,16 +137,35 @@ def execute(self, node_dict: Dict[IdentityHash, "ExecNode"]) -> Optional[Any]:
 
         if self.executed:
             logger.debug(f"Skipping execution of a pre-computed node {self.id}")
+
+            # reset the profiling
+            # (for example setup ExecNodes have a profiling on the 1st execution
+            #   but afterwards their profiling should be reset)
+            if Cfg.TAWAZI_PROFILE_ALL_NODES:
+                self.profile = PROFILE_ZERO
+
             return self.result
-        # 1. prepare args and kwargs for usage:
+
+        # 1. pre-
+        # 1.1 prepare the profiling
+        if Cfg.TAWAZI_PROFILE_ALL_NODES:
```

Comment:
couldn't we implement a context manager instead of two ifs ? It would be cleaner + you do a `pre` and `post` so I think it would be the perfect case

Reply:
actually I though of the same thing... and the context manager already exists in Profile

Reply:
yeah that's why

Reply:
Done. Can you review it please ?